File: example_scripts/index_example.py
import pandas as pd
from onboard.client import OnboardClient
from onboard.client.dataframes import points_df_from_streaming_timeseries
from onboard.client.models import PointSelector, TimeseriesQuery, PointData
from datetime import datetime, timezone, timedelta
from typing import List
import pytz
from key import api_key

client = OnboardClient(api_key=api_key)

print(list(pd.DataFrame(client.get_all_buildings())['name'])) # returns list of buildings that you have access to (you may not have 'Laboratory' in your set)

query = PointSelector()
query.point_types     = ['Zone Temperature'] # can list multiple point
query.equipment_types = ['fcu']              # types, equipment types,
query.buildings       = ['Laboratory']       # buildings, etc.
selection = client.select_points(query)

start = pd.Timestamp("2022-03-29 00:00:00", tz="utc")
end = pd.Timestamp("2022-07-29 00:00:00", tz="utc")

timeseries_query = TimeseriesQuery(point_ids = selection['points'], start = start, end = end)
sensor_data = points_df_from_streaming_timeseries(client.stream_point_timeseries(timeseries_query))

File: example_scripts/index_example.R
install.packages(c('OnboardClient', 'lubridate'))          # install whatever libraries you don't already have
library(OnboardClient)
library(lubridate)                                      # for datetime handling

api.setup()                                             # will prompt for api key
query <- PointSelector()                                # create point selector
query$point_types       <- c('Zone Temperature')        # can list multiple points
query$equipment_types   <- c('fcu')                     # types, equipment types,
query$buildings         <- c('Laboratory')               # buildings, etc.
selection <- select_points(query)

start <- as_datetime("2022-03-29 00:00:00", tz = "UTC")
end <- as_datetime("2022-07-29 00:00:00", tz = "UTC")

sensor_data <- get_timeseries(start_time = start, end_time = end, point_ids = selection$points) #Queries timeseries data for the selection list we got above